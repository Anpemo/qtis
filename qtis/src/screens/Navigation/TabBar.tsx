import React, { useState } from 'react'
import { View, StyleSheet, Dimensions } from 'react-native'
import { COLORS, SHADOW } from '../../../constants'
import Tab from '../Navigation/Tab'

const { width } = Dimensions.get('screen')

const styles = StyleSheet.create({
  wrapper: {
    position: 'absolute',
    bottom: 1,
    width,
    alignItems: 'center',
    justifyContent: 'center'
  },
  container: {
    flexDirection: 'row',
    backgroundColor: COLORS.white,
    width: '100%',
    justifyContent: 'space-between',
    ...SHADOW
  }
})

function TabBar ({ state, navigation }: any) {
  const [selected, setSelected] = useState('Browser')
  const { routes } = state
  const renderColor = (currentTab: any) => currentTab === selected ? COLORS.brown : 'black'

  function handlePress (activeTab: any, index: any) {
    setSelected(activeTab)
    if (state.index !== index) {
      navigation.navigate(activeTab)
    }
  }
  return (
    <View style={styles.wrapper}>
        <View style={styles.container}>
            {
                routes.map((route: any, index: any) =>
                 <Tab
                    tab={route}
                    icon={route.params.icon}
                    onPress={() => handlePress(route.name, index)}
                    color={renderColor(route.name)}
                    key={route.key}
                />)
            }
        </View>
    </View>)
}

export default TabBar
