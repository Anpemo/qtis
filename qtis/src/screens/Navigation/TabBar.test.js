import React from 'react'
import TabBar from './TabBar'
import { cleanup, fireEvent, render } from '@testing-library/react-native'

describe('Given a TabBar component', () => {
  let component
  const navigate = jest.fn()
  const routes = [
    { name: 'Browser', params: { icon: 'search1' }, key: 1 },
    { name: 'Scanner', params: { icon: 'barcode-scan' }, key: 2 },
    { name: 'Profile', params: { icon: 'user' }, key: 3 }
  ]
  beforeEach(() => {
    component = (
                  <TabBar navigation={{ navigate }} state={{ routes }} />
    )
  })
  afterEach(cleanup)
  describe('When calling it', () => {
    test('Then it will match the snapshot', () => {
      const tabBar = render(component)
      expect(tabBar).toMatchSnapshot()
    })
    test('Then it will render a search', () => {
      const { getAllByTestId } = render(component)
      const tabID = getAllByTestId('tabContainer')[0]
      fireEvent.press(tabID)
      expect(tabID).toHaveBeenCalled()
    })
  })
})
