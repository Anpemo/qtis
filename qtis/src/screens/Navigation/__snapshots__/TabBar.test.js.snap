// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a TabBar component When calling it Then it will match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "bottom": 1,
      "justifyContent": "center",
      "position": "absolute",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "elevation": 4,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.23,
        "shadowRadius": 2.62,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 5,
        }
      }
      testID="tabContainer"
    >
      <Text />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 5,
        }
      }
      testID="tabContainer"
    >
      <Text />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 5,
        }
      }
      testID="tabContainer"
    >
      <Text />
    </View>
  </View>
</View>
`;
